from_test.go

```diff
@@ -51,6 +58,20 @@ func TestFromChannel(t *testing.T) {
 	}
 }
 
+func TestFromTypedChannel(t *testing.T) {
+	c := make(chan int, 3)
+	c <- 10
+	c <- 15
+	c <- -3
+	close(c)
```

Comment:
I actually have no idea how recv on a closed chan works below.
But it seemed like it worked so far before as well lol.

Reply:
I expect `Recv` to behave well, since it is the reflection way to do `<-` operator.